allure-spock/src/main/java/io/qameta/allure/spock/AllureSpock.java

```diff
@@ -308,4 +329,49 @@ public void afterIteration(final IterationInfo iteration) {
     public AllureLifecycle getLifecycle() {
         return lifecycle;
     }
+
+    private void writeBlocks(final FeatureInfo feature, final IterationInfo iteration) {
+        String blockText = StringUtils.EMPTY;
+        for (BlockInfo block: feature.getBlocks()) {
+            if (!isEmptyOrContainsOnlyEmptyStrings(block.getTexts())) {
+                final int size = block.getTexts().size();
+                for (int i = 0; i < size; i++) {
+                    if (iteration != null) {
+                        blockText = generationParams(block.getTexts().get(i), feature.getDataVariables(), iteration);
+                    }
+                    final String kind = i == 0 ? block.getKind().name() : "and";
+                    writeStep(kind, blockText);
+                }
+            } else {
+                writeStep(block.getKind().name(), "-----");
+            }
+        }
+    }
+
+    private void writeStep(final String kind, final String data) {
+        final String kindText = stepSpockMap.get(kind.toUpperCase());
+        final StringBuffer stringBuilder = new StringBuffer();
+        stringBuilder
+                .append(kindText)
+                .append('\t').append(data);
+        Allure.step(stringBuilder.toString());
+    }
+
+    private String generationParams(
+            final String input, final List<String> dataVariables, final IterationInfo iteration) {
+        final FeatureInfo tempFeature = new FeatureInfo();
+        tempFeature.setName(input);
+        for (String variable : dataVariables) {
+            tempFeature.addParameterName(variable);
+        }
+        tempFeature.setIterationNameProvider(new UnrollNameProvider(tempFeature, input));
+        return tempFeature.getIterationNameProvider().getName(iteration);
+    }
+
+    private boolean isEmptyOrContainsOnlyEmptyStrings(final List<String> strings) {
+        final boolean countOnlyEmptyStrings = strings
+                .stream()
+                .noneMatch(StringUtils::isNotBlank);
+        return CollectionUtils.isEmpty(strings) || countOnlyEmptyStrings;
```

Comment:
you introduce dependency only for one method? Why not to use `Objects.isNull(strings) || strings.isEmpty() || ...` instead?

Reply:
Yes, I made an dependency by discussing with Eroshenko